content/blog/2018/app-metrics-istio/index.md

```diff
@@ -0,0 +1,82 @@
+
+A simpler approach is to scrape the application’s metrics  using Istio’s Prometheus directly and that’s what I want to focus on here. 
+
+## Sending application metrics
+To send custom metrics from your application, you need to instrument your application using Prometheus’ [client libraries](https://prometheus.io/docs/instrumenting/clientlibs/). Which  library to use depends on the language you’re using. As a C#/.NET developer, I used the  [.NET client](https://github.com/prometheus-net/prometheus-net) for Prometheus and [this blog post](https://www.olivercoding.com/2018-07-22-prometheus-dotnetcore/) from Daniel Oliver has step-by-step instructions on how to send custom metrics from an ASP.NET Core application and see them in a local Prometheus server.
+
+One thing you need to pay attention to is the port where you’re exposing your Prometheus metrics. In ASP.NET Core, the default port is **5000**. When running locally, application metrics are exposed on `localhost:5000/metrics`. However, when you containerize your app, it’s common to expose your application over a different port, like **8080**, and this becomes relevant later when we talk about configuration.
+
+Assuming that you containerized and deployed your application on an Istio-enabled Kubernetes cluster, let’s now take a look at what we need to do to get these application metrics scraped by Istio’s Prometheus.
+
+## Configuration
+In Istio 1.0.5, the default installation files for Kubernetes, `istio-demo.yaml` or `istio-demo-auth.yaml`, already have scraping configurations for Prometheus under a `ConfigMap`. You can just search for `prometheus.yml`. There are 2 scraping jobs that are relevant for application metrics: 
+
+{{< text bash >}}
+   - job_name: 'kubernetes-pods'
+      kubernetes_sd_configs:
+      - role: pod
+...
+   - job_name: 'kubernetes-pods-istio-secure'
+      scheme: https
+
+{{< /text >}}
+
+These are the jobs that scrape metrics from regular pods and pods where mTLS is enabled. It looks like Istio’s Prometheus should automatically scrape application metrics. However, in my first try, it didn’t work. I wasn’t sure what was wrong but Prometheus has some default endpoints:
+* `/config`: to see the current configuration of Prometheus.
+* `/metrics`: to see the scraped metrics.
+* `/targets`: to see the targets that’s being scraped and their status. 
+
+All of these endpoints are useful for debugging Prometheus:
+
+{{< image width="60%" ratio="43%" link="./image1.png" caption="Prometheus Targets page" >}}
+
+Turns out, I needed to add some annotations in my pod YAML files in order to get Prometheus scrape the pod. I had to tell Prometheus to scrape the pod and on which port with these annotations:
+
+{{< text bash >}}
+ template:
```

Comment:
Make sure that the first level of your code is on the same level as the `{{< text yaml >}} tag to avoid linter issues.
```suggestion
template:
```